### PR TITLE
feat(genQa): hide toggle button by default

### PR DESCRIPTION
**Pull request by jelmedini**
[SVCC-3649](https://coveord.atlassian.net/browse/SVCC-3649)

- Hide Toggle button of the generated answer by default.
- Add Toggle button as an option to allow the user to choose whether to use this feature or not.

When `with-toggle` option is NOT provided:
Screenshot 2024-05-02 at 10 14 14 AM

When `with-toggle` option is provided:
Screenshot 2024-05-02 at 10 14 40 AM

https://github.com/coveo/ui-kit/assets/119955059/cc901a21-6df2-4e00-8bd8-030780ac69e7

[SVCC-3649]: https://coveord.atlassian.net/browse/SVCC-3649?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dmgauthier — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

**Review by lvu285 — APPROVED** [review]
tested, looks good on my side

**Review by lbergeron — COMMENTED** [review]
Can you please confirm if the following sequence works correctly?

1. The search page shows the toggle switch.
2. A visitor performs a query, then puts the toggle switch in the OFF position.
3. The search page is updated to hide the toggle switch.
4. When the visitor comes back, are new generated answers shown to the end user?

I figure new answers might not be visible considering the search page remembers the toggle switch position across visits.

File: packages/atomic/src/components/search/atomic-generated-answer/atomic-generated-answer.tsx [hunk 1]
```diff
@@ -81,6 +81,11 @@ export class AtomicGeneratedAnswer implements InitializableComponent {
    */
   @Prop() answerStyle: GeneratedAnswerStyle = 'default';
 
+  /**
+   * Whether to allow the answer to be shown or hidden using the toggle button
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/search/atomic-generated-answer/atomic-generated-answer.tsx:85]:
```suggestion
   * Whether to render a toggle button that lets the user hide or show the answer.
```

**Review by fbeaudoincoveo — APPROVED** [review]

**Review by fpbrault — APPROVED** [review]

**Comment by jelmedini** [comment]
> Can you please confirm if the following sequence works correctly?
> 
> 1. The search page shows the toggle switch.
> 2. A visitor performs a query, then puts the toggle switch in the OFF position.
> 3. The search page is updated to hide the toggle switch.
> 4. When the visitor comes back, are new generated answers shown to the end user?
> 
> I figure new answers might not be visible considering the search page remembers the toggle switch position across visits.

Indeed this was an issue, implemented the fix that we discussed aside

**Pull request merged**